package.json

```diff
@@ -44,20 +44,24 @@
     }
   ],
   "devDependencies": {
+    "@defi-wonderland/smock": "^2.3.5",
     "@eth-optimism/contracts": "^0.5.37",
-    "@nomicfoundation/hardhat-ethers": "^3.0.4",
     "@nomiclabs/hardhat-ethers": "^2.2.1",
     "@nomiclabs/hardhat-etherscan": "^3.1.7",
     "@nomiclabs/hardhat-waffle": "^2.0.3",
-    "@openzeppelin/hardhat-upgrades": "^2.2.1",
+    "@openzeppelin/hardhat-upgrades": "^1.28.0",
     "@size-limit/preset-small-lib": "^7.0.8",
-    "@typechain/hardhat": "^5.0.0",
+    "@typechain/ethers-v5": "^11.1.1",
+    "@typechain/hardhat": "^6.1.6",
     "@types/chai": "^4.3.6",
     "@types/dotenv": "^8.2.0",
+    "@types/lodash": "^4.14.199",
     "@types/lodash.get": "^4.4.7",
     "@typescript-eslint/eslint-plugin": "^6.6.0",
     "@typescript-eslint/parser": "^6.6.0",
+    "@uma/financial-templates-lib": "^2.33.0",
```

Comment:
Would be really nice to remove this, why exactly do we need it?

Reply:
Done